tests/test_server.py

```diff
@@ -750,6 +750,28 @@ def test_load_model(rasa_app: SanicTestClient, trained_core_model):
     assert old_fingerprint != response.json["fingerprint"]
 
 
+def test_load_unpacked_model(rasa_app: SanicTestClient, trained_core_model):
+    _, response = rasa_app.get("/status")
+
+    assert response.status == 200
+    assert "fingerprint" in response.json
+
+    old_fingerprint = response.json["fingerprint"]
+
+    with unpack_model(trained_core_model) as unpacked_model:
+        data = {"model_file": unpacked_model}
```

Comment:
I think we should support a new json parameter here, e.g. `model_folder`. we can leave the loading unchanged in `Agent` but get the path in the endpoint with something like

```python
rjs = request.json
model_path = rjs.get("model_file") or rjs.get("model_folder")
```

Reply:
Do you have any thoughts on handling such parameter interdependencies in the HTTP API spec? I came across this unresolved discussion here https://github.com/OAI/OpenAPI-Specification/issues/256, but don't have any strong preferences of my own yet.